From 0ee1f1b52152e395e78906a02f41db4435422aae Mon Sep 17 00:00:00 2001
From: Peng Yi <pengyi37@huawei.com>
Date: Thu, 9 Mar 2023 11:02:40 +0800
Subject: [PATCH 001/921] [Huawei] bridge: switchdev: Fix abi broken:
 switchdev_attr_id

Offering: HULK
EulerOS inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I6LJTI

--------------------------------

When adding enum SWTICHDEV_ATTR_ID_BRIDGE_VLAN_PROTOCOL
to switchdev_attr_id, part of macro values are changed,
causing problems when the value are passed-by to the
third-part ko compiled by old version of header file.
We move the macro at the end of the enum class, so
it will not affect other values.

Fixes: 89a7f155e6b2 (bridge: switchdev: Notify about VLAN protocol changes)
Signed-off-by: Peng Yi <pengyi37@huawei.com>
---
 include/net/switchdev.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/net/switchdev.h b/include/net/switchdev.h
index 94bb6e055589..1b7dc1435cf3 100644
--- a/include/net/switchdev.h
+++ b/include/net/switchdev.h
@@ -39,12 +39,12 @@ enum switchdev_attr_id {
 	SWITCHDEV_ATTR_ID_PORT_MROUTER,
 	SWITCHDEV_ATTR_ID_BRIDGE_AGEING_TIME,
 	SWITCHDEV_ATTR_ID_BRIDGE_VLAN_FILTERING,
-	SWITCHDEV_ATTR_ID_BRIDGE_VLAN_PROTOCOL,
 	SWITCHDEV_ATTR_ID_BRIDGE_MC_DISABLED,
 	SWITCHDEV_ATTR_ID_BRIDGE_MROUTER,
 #if IS_ENABLED(CONFIG_BRIDGE_MRP)
 	SWITCHDEV_ATTR_ID_MRP_PORT_ROLE,
 #endif
+	SWITCHDEV_ATTR_ID_BRIDGE_VLAN_PROTOCOL
 };
 
 struct switchdev_attr {
-- 
2.17.1

